source/_docs/z-wave/adding.markdown

```diff
@@ -3,81 +3,58 @@ title: "Z-Wave Devices - Adding and Removing"
-```
+If your device isn't responding to this process, possibly because you've factory reset it or it has failed, you can remove it using **Remove Failed Node**.
 
-(remember you can run this in Home Assistant OS or Supervised, you can use the Terminal Add-on)
+## Troubleshooting
 
-<div class='note warning'>
-Ensure you keep a backup of this key. If you have to rebuild your system and don't have a backup of this key, you won't be able to reconnect to any security devices. This may mean you have to do a factory reset on those devices, and your controller, before rebuilding your Z-Wave network.
-</div>
+Problems adding or removing devices?
 
-## Adding Secure Devices
+### Already Added
 
-After defining your network key, follow these steps to add (include) a secure Z-Wave device:
+If your device was previously added to another controller but not removed from it, **you must remove it before adding it to Home Assistant**.
 
-1. Go to the [Z-Wave control panel](/docs/z-wave/control-panel/) in the Home Assistant frontend
-2. Click the **Add Node Secure** button in the *Z-Wave Network Management* card - this will place the controller in inclusion mode
-3. Activate your device to be included by following the instructions provided with the device
-4. With the device in its final location, run a *Heal Network*
+1. **Remove the device from the old controller**, if possible. If your device was added to a different system (SmartThings, Wink, etc.) follow that system's instructions to remove the device. Then try adding the device to Home Assistant again.
+2. If you can't remove it from the old controller, **try removing the device using Home Assistant**. Follow the instructions in [Removing Devices](#removing-devices); even though it hasn't been added to Home Assistant yet, most Z-Wave devices will respond to exclusion mode from *any* controller. Then try adding the device to Home Assistant again. (Secure devices with a theft protection feature may not allow this.)
+3. If removing the device doesn't help, **try a factory reset** as a last resort. Check your device's manual for instructions.
 
-<div class='note warning'>
-Secure devices require additional bandwidth, and too many secure devices can slow down your Z-Wave network. We recommend only using secure inclusion for devices that require it, such as locks.
-</div>
+### Range Issues
 
-## Removing Devices
+Normally you can add and remove devices as long as they are within range of any Z-Wave Plus device in your network, using a feature called [network-wide inclusion](https://z-wavealliance.org/z-wave_plus_certification/).
 
-To remove (exclude) a Z-Wave device from your system:
+If you are using classic Z-Wave devices, your device may need to be within range of your [controller](/docs/z-wave/controllers/) to be added or removed. If you cannot move the device (e.g. wall switches), you can try temporarily relocating your controller to be near your device.
```

Comment:
I'd change "classic Z-Wave devices" to "older non-ZWave Plus model devices"

Reply:
That's a good suggestion! On it.